code/game/camera_manager/camera_manager.dm

```diff
@@ -0,0 +1,224 @@
+		var/atom/movable/screen/plane_master/instance = new plane()
+		add_plane(instance)
+
+/datum/component/camera_manager/Destroy(force, ...)
+	. = ..()
+	range_turfs = null
+	current_area = null
+	cam_plane_masters = null
+	QDEL_NULL(cam_background)
+	QDEL_NULL(cam_screen)
+
+/datum/component/camera_manager/proc/add_plane(atom/movable/screen/plane_master/instance)
+	instance.assigned_map = map_name
+	instance.del_on_map_removal = FALSE
+	if(instance.blend_mode_override)
+		instance.blend_mode = instance.blend_mode_override
+	instance.screen_loc = "[map_name]:CENTER"
+	cam_plane_masters["[instance.plane]"] = instance
+
+/datum/component/camera_manager/proc/register(source, mob/user)
+	var/client/user_client = user.client
+	if(!user_client)
+		return
+	user_client.register_map_obj(cam_background)
+	user_client.register_map_obj(cam_screen)
+	for(var/plane_id in cam_plane_masters)
+		user_client.register_map_obj(cam_plane_masters[plane_id])
+
+/datum/component/camera_manager/proc/unregister(source, mob/user)
+	var/client/user_client = user.client
+	if(!user_client)
+		return
+	user_client.clear_map(cam_background)
+	user_client.clear_map(cam_screen)
+	for(var/plane_id in cam_plane_masters)
+		user_client.clear_map(cam_plane_masters[plane_id])
+
+/datum/component/camera_manager/RegisterWithParent()
+	. = ..()
+	START_PROCESSING(SSdcs, src)
+	SEND_SIGNAL(parent, COMSIG_CAMERA_MAPNAME_ASSIGNED, map_name)
+	RegisterSignal(parent, COMSIG_CAMERA_REGISTER_UI, PROC_REF(register))
+	RegisterSignal(parent, COMSIG_CAMERA_UNREGISTER_UI, PROC_REF(unregister))
+	RegisterSignal(parent, COMSIG_CAMERA_SET_NVG, PROC_REF(enable_nvg))
+	RegisterSignal(parent, COMSIG_CAMERA_CLEAR_NVG, PROC_REF(disable_nvg))
+	RegisterSignal(parent, COMSIG_CAMERA_SET_AREA, PROC_REF(set_camera_rect))
+	RegisterSignal(parent, COMSIG_CAMERA_SET_TARGET, PROC_REF(set_camera))
+	RegisterSignal(parent, COMSIG_CAMERA_CLEAR, PROC_REF(clear_camera))
+
+/datum/component/camera_manager/UnregisterFromParent()
+	. = ..()
+	STOP_PROCESSING(SSdcs, src)
+
+	UnregisterSignal(parent, COMSIG_CAMERA_REGISTER_UI)
+	UnregisterSignal(parent, COMSIG_CAMERA_UNREGISTER_UI)
+	UnregisterSignal(parent, COMSIG_CAMERA_SET_NVG)
+	UnregisterSignal(parent, COMSIG_CAMERA_CLEAR_NVG)
+	UnregisterSignal(parent, COMSIG_CAMERA_SET_AREA)
+	UnregisterSignal(parent, COMSIG_CAMERA_SET_TARGET)
+	UnregisterSignal(parent, COMSIG_CAMERA_CLEAR)
+
+/datum/component/camera_manager/proc/clear_camera()
+	current_area = null
+	current = null
+	target_x = null
+	target_y = null
+	target_z = null
+	target_width = null
+	target_height = null
+	show_camera_static()
+
+/datum/component/camera_manager/proc/set_camera(source, atom/target, w, h)
+	render_mode = RENDER_MODE_TARGET
+	if(current)
+		UnregisterSignal(current, COMSIG_PARENT_QDELETING)
+	current = target
+	target_width = w
+	target_height = h
+	RegisterSignal(current, COMSIG_PARENT_QDELETING, PROC_REF(show_camera_static))
```

Comment:
This frequently throws `parent_qdeleting overridden. Use override = TRUE to suppress this warning` when pressing any of the target buttons on the right side.

Reply:
Discussed on discord: This occurs when you press a button that isn't a flare, then go to one that is a flare that you previously viewed.

Reply:
This should now be resolved